<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 3 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [1]:
import matplotlib.pyplot as plt
import numpy as np
#import seaborn as sns
import pandas as pd
from scipy.stats import ttest_ind, ttest_ind_from_stats, ttest_rel ,ttest_1samp


'''pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)'''


"pd.set_option('display.max_rows', 500)\npd.set_option('display.max_columns', 500)"

In [2]:
column_header=[ 
    'party','handicapped-infants' , 'water-project-cost-sharing', 'adoption-of-the-budget-resolution', 'physician-fee-freeze','el-salvador-aid' , 'religious-groups-in-schools' ,
    'anti-satellite-test-ban', 'aid-to-nicaraguan-contras' ,'mx-missile', 'immigration','synfuels-corporation-cutback', 'education-spending' , 'superfund-right-to-sue',
    'crime', 'duty-free-exports' ,'export-administration-act-south-africa']


df = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data',names=column_header,header=1)

In [3]:

df = df.replace({
    'y':1,
    'n':0,
    '?':np.nan
})

df.tail(5)
rep = df[df['party'] == 'republican']
dem = df[df['party'] == 'democrat']
df.head()

party  handicapped-infants  water-project-cost-sharing  \
0  democrat                  NaN                         1.0   
1  democrat                  0.0                         1.0   
2  democrat                  1.0                         1.0   
3  democrat                  0.0                         1.0   
4  democrat                  0.0                         1.0   

   adoption-of-the-budget-resolution  physician-fee-freeze  el-salvador-aid  \
0                                1.0                   NaN              1.0   
1                                1.0                   0.0              NaN   
2                                1.0                   0.0              1.0   
3                                1.0                   0.0              1.0   
4                                0.0                   1.0              1.0   

   religious-groups-in-schools  anti-satellite-test-ban  \
0                          1.0                      0.0   
1                          1.0                      0.0   
2                          1.0                      0.0   
3                          1.0                      0.0   
4                          1.0                      0.0   

   aid-to-nicaraguan-contras  mx-missile  immigration  \
0                        0.0         0.0          0.0   
1                        0.0         0.0          0.0   
2                        0.0         0.0          0.0   
3                        0.0         0.0          0.0   
4                        0.0         0.0          0.0   

   synfuels-corporation-cutback  education-spending  superfund-right-to-sue  \
0                           1.0                 0.0                     1.0   
1                           1.0                 0.0                     1.0   
2                           1.0                 NaN                     1.0   
3                           0.0                 0.0                     1.0   
4                           0.0                 0.0                     NaN   

   crime  duty-free-exports  export-administration-act-south-africa  
0    1.0                0.0                                     0.0  
1    0.0                0.0                                     1.0  
2    1.0                1.0                                     1.0  
3    1.0                1.0                                     1.0  
4    1.0                1.0                                     1.0

In [6]:
ttest_ind(dem['crime'],rep['crime'], nan_policy='omit')

Ttest_indResult(statistic=masked, pvalue=masked_array(data=--,
             mask=True,
       fill_value=1e+20,
            dtype=float64))

Null: The mean Dem support == The mean Rep support

Alternitive: The mean Dem support != The mean Rep support

Results: Givien the P value is 3.2341 e-46 the Null Hypothesis is regected.

Democrats suport this bill more than the Republicans. 

In [5]:
ttest_ind(dem['education-spending'],rep['education-spending'], nan_policy='omit')

Ttest_indResult(statistic=-20.32755974262684, pvalue=1.3125167286634945e-63)

Null: The mean Dem support == The mean Rep support

Alternitive: The mean Dem support != The mean Rep support

Results: Givien the P value is 1.3125 e-63 the Null Hypothesis is regected.

Republicans suport this bill more than the Democrats.


In [15]:
ttest_ind(dem['water-project-cost-sharing'],rep['water-project-cost-sharing'], nan_policy='omit')


Ttest_indResult(statistic=0.03972999357457669, pvalue=0.9683290905303186)

Null: The mean Dem support == The mean Rep support

Alternitive: The mean Dem support != The mean Rep support

Results: Givien the P value is .968 the Null Hypothesis is NOT regected.

Republicans suport this bill as much as Democrats with confidence interval of 96%